openpype/hosts/max/plugins/publish/validate_loaded_plugin.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+"""Validator for Loaded Plugin."""
+import os
+from pyblish.api import InstancePlugin, ValidatorOrder
+from pymxs import runtime as rt
+
+from openpype.pipeline.publish import (
+    RepairAction,
+    OptionalPyblishPluginMixin,
+    PublishValidationError
+)
+from openpype.hosts.max.api.lib import get_plugins
+
+
+class ValidateLoadedPlugin(OptionalPyblishPluginMixin,
+                           InstancePlugin):
+    """Validates if the specific plugin is loaded in 3ds max.
+    Studio Admin(s) can add the plugins they want to check in validation
+    via studio defined project settings
+    If families = ["*"], all the required plugins would be validated
+    If families
+
+    """
+
+    order = ValidatorOrder
+    hosts = ["max"]
+    label = "Validate Loaded Plugins"
+    optional = True
+    actions = [RepairAction]
+
+    def get_invalid(self, instance):
+        """Plugin entry point."""
+        if not self.is_active(instance.data):
+            self.log.debug("Skipping Validate Loaded Plugin...")
+            return
+
+        required_plugins = (
+            instance.context.data["project_settings"]["max"]["publish"]
+                                 ["ValidateLoadedPlugin"]
+                                 ["family_plugins_mapping"]
+        )
+
+        if not required_plugins:
+            return
+
+        invalid = []
+
+        # get all DLL loaded plugins in Max and their plugin index
+        available_plugins = {
+            plugin_name.lower(): index for index, plugin_name in enumerate(
+                get_plugins())
+        }
+
+        for families, plugin in required_plugins.items():
+            # Out of for loop build the instance family lookup
+            instance_families = {instance.data["family"]}
+            instance_families.update(instance.data.get("families", []))
+            self.log.debug(f"{instance_families}")
+            # In the for loop check whether any family matches
+            match_families = {fam.strip() for fam in
+                              families.split(",") if fam.strip()}
+            self.log.debug(f"match_families: {match_families}")
+            has_match = "*" in match_families or match_families.intersection(
+                instance_families) or families == "_"
+            if not has_match:
+                continue
```

Comment:
```suggestion

        # Build instance families lookup
        instance_families = {instance.data["family"]}
        instance_families.update(instance.data.get("families", []))
        self.log.debug(f"Checking plug-in validation for instance families: {instance_families}")
        for families, plugin in required_plugins.items():
        
            # Check for matching families
            match_families = {fam.strip() for fam in
                              families.split(",") if fam.strip()}
            self.log.debug(f"Plug-in family requirements: {match_families}")
            has_match = "*" in match_families or match_families.intersection(
                instance_families) or families == "_"
            if not has_match:
                continue
```

I still don't understand the check against families `_` - what's that about?

Reply:
I am doing this because it can check all the plugins for user preferences even if the families are not *. But I think if I adopt the structure you suggested [above](https://github.com/ynput/OpenPype/pull/5820#pullrequestreview-1714654139), I dont really need to check against families `_`, just need to add condition if the key value of the families are None and there is plugin name of which users want to check, it will check across all the families.